scripts/dependency_install_functions.sh

```diff
@@ -64,6 +64,22 @@ else
   fi
 fi
 
+install_python() {
+  if [ ! -e "${DATAFED_DEPENDENCIES_INSTALL_PATH}/.python_installed-${DATAFED_PYTHON_VERSION}" ]; then
+    # Check if the deadsnakes repository has already been added to avoid issues with gpg
+    if ! grep -qr '^deb .\+deadsnakes' /etc/apt/sources.list.d/; then
+	"$SUDO_CMD" apt update
+	"$SUDO_CMD" apt install -y software-properties-common
+	"$SUDO_CMD" add-apt-repository -y ppa:deadsnakes/ppa
+	"$SUDO_CMD" apt update
+    fi
+
+    "$SUDO_CMD" apt install -y "python${DATAFED_PYTHON_VERSION}" "python${DATAFED_PYTHON_VERSION}-dev" "python${DATAFED_PYTHON_VERSION}-venv" "python${DATAFED_PYTHON_VERSION}-distutils"
+
+    touch "${DATAFED_DEPENDENCIES_INSTALL_PATH}/.python_installed-${DATAFED_PYTHON_VERSION}"
+  fi
+}
+
```

Comment:
We probably need a way to install all dependencies that is not distro specific, I really don't like using apt for this it makes things brittle.